src/datatypes/src/value.rs

```diff
@@ -303,6 +325,16 @@ pub fn timestamp_to_scalar_value(unit: TimeUnit, val: Option<i64>) -> ScalarValu
     }
 }
 
+/// Cast the 64-bit elapsed time into the arrow ScalarValue by time unit.
+pub fn time_to_scalar_value(unit: TimeUnit, val: Option<i64>) -> ScalarValue {
+    match unit {
+        TimeUnit::Second => ScalarValue::Time32Second(val.map(|i| i as i32)),
```

Comment:
Can we return an error when time unit is Second/Millisecond but converting `val` to i32 overflows instead of casting to i32 silently?